common/src/main/java/com/wynntils/features/overlays/CustomPlayerListFeature.java

```diff
@@ -0,0 +1,289 @@
+            RenderUtils.drawTexturedRect(
+                    poseStack,
+                    Texture.PLAYER_INFO_OVERLAY.resource(),
+                    getRenderX() + ROLL_WIDTH,
+                    getRenderY(),
+                    0,
+                    Texture.PLAYER_INFO_OVERLAY.width() - ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    ROLL_WIDTH,
+                    0,
+                    Texture.PLAYER_INFO_OVERLAY.width() - ROLL_WIDTH,
+                    Texture.PLAYER_INFO_OVERLAY.height(),
+                    Texture.PLAYER_INFO_OVERLAY.width(),
+                    Texture.PLAYER_INFO_OVERLAY.height());
+        }
+
+        @Override
+        protected void onConfigUpdate(ConfigHolder configHolder) {
+            animationPercentage.setOpeningDuration(Duration.of(openingDuration.get(), ChronoUnit.MILLIS));
+        }
+    }
+
+    private static final class AnimationPercentage {
+        private final Supplier<Boolean> openingContitions;
+        private Duration openingDuration;
+        private Instant lastTime = Instant.EPOCH;
+
+        private double openingProgress;
+
+        private AnimationPercentage(Supplier<Boolean> openingContitions, Duration openingDuration) {
+            this.openingContitions = openingContitions;
+            this.openingDuration = openingDuration;
+        }
+
+        private double getAnimation() {
+            if (openingDuration.isZero() || openingDuration.isNegative()) return 1;
+            if (openingProgress == 0) {
+                lastTime = Instant.now().minus(1, ChronoUnit.MILLIS);
+            }
+
+            if (openingContitions.get()) {
+                addOpeningProgress(
+                        Duration.between(lastTime, Instant.now()).toMillis() / ((double) openingDuration.toMillis()));
+            } else if (openingProgress > 0) {
+                addOpeningProgress(
+                        -Duration.between(lastTime, Instant.now()).toMillis() / ((double) openingDuration.toMillis()));
+            }
+            lastTime = Instant.now();
+            return applyTransformation(openingProgress);
+        }
+
+        public void addOpeningProgress(double openingProgress) {
+            setOpeningProgress(this.openingProgress + openingProgress);
+        }
+
+        public void setOpeningProgress(double openingProgress) {
+            this.openingProgress = Math.clamp(0, 1, openingProgress);
+        }
+
+        private static double applyTransformation(double openingProgress) {
+            return java.lang.Math.sin((float) (openingProgress / 2f * java.lang.Math.PI));
+        }
+
+        public void setOpeningDuration(Duration openingDuration) {
+            this.openingDuration = openingDuration;
+        }
+
+        public boolean finishedClosingAnimation() {
+            return openingProgress == 0;
+        }
+    }
+
+    private static final class TimelimitedExecutioner<T> {
+
```

Comment:
```suggestion
```

I really don't like extra empty lines after classes, I wish spotless would clean it up

Reply:
I agree. Finally got around to fixing it: #1779.

Reply:
@Mcrtin That means that if you merge in the latest main and run spotlessApply, this will be fixed automatically.